File: Team1/Riya Tanwar/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int len = nums.length;
        int[] temp = new int[len];
        Arrays.fill(temp, -1);
        for(int i=0;i<len;i++){
            temp[(i+k)%len] = nums[(i+k)%len];
            nums[(i+k)%len] = temp[i]!=-1 ? temp[i]:nums[i];
        }
    }
}

File: Team4/Nikita/RotateArray.cpp
class Solution {
public:
  void rotate(vector<int>& nums, int k) {
    
    k = k%(nums.size());
    Reverse(nums,nums.size()-k,nums.size()-1);
    Reverse(nums,0,nums.size()-k-1);
    Reverse(nums,0,nums.size()-1);   
       }
      
   void Reverse(vector<int>& nums, int l , int h){
        while(l<h){
            int t= nums[l];
        nums[l]=nums[h];
        nums[h]=t;
        l++;
        h--;}
    }
}; 
